internal/cloudapi/server.go

```diff
@@ -331,6 +331,7 @@ func (h *apiHandlers) Compose(ctx echo.Context) error {
 			t := target.NewAWSS3Target(&target.AWSS3TargetOptions{
 				Filename:        imageType.Filename(),
 				Region:          awsS3UploadOptions.Region,
+				Endpoint:        *awsS3UploadOptions.Endpoint,
```

Comment:
This will dereference `nil` if the endpoint isn't specified (which will be most of the time).  It'll need something similar to [the session token handling](https://github.com/osbuild/osbuild-composer/pull/1666/files#diff-b4cd771217e68c4d9daf38fb95f1526891f973015d00a94d071f5282617959bdR288-R312).